packaging/deb/build.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+set -e
+
+LOCAL_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+SRC_DIR=$(dirname "$(dirname "$LOCAL_DIR")")
+cd $SRC_DIR
+
+################################################################################
+### Check for basic requirements
+################################################################################
+
+if ! which git >/dev/null 2>&1; then
+    echo "git not installed"
+    exit 1
+fi
+if ! git rev-parse >/dev/null 2>&1; then
+    echo "not a git repository"
+    exit 1
+fi
+if [ "$(git rev-parse --show-toplevel)" != "$SRC_DIR" ]; then
+    echo "$SRC_DIR is not a git repository"
+    exit 1
+fi
+if ! which python >/dev/null 2>&1; then
+    echo "python not installed"
+    exit 1
+fi
+if ! git diff-index --quiet HEAD >/dev/null 2>&1; then
+    echo "git index is dirty - either stash or commit your changes"
+    exit 1
+fi
+if ! which docker >/dev/null 2>&1; then
+    echo "docker not installed"
+    exit 1
+fi
+
+################################################################################
+# Read envvars
+################################################################################
+
+if [ -z "$DOCKER_BASE" ]; then
+    echo "DOCKER_BASE is a required environment variable"
+    exit 1
+fi
+echo "DOCKER_BASE: $DOCKER_BASE"
+DOCKER_BUILD_ID="caffe-nv-debuild"
+DOCKER_BASE_TAG="caffe-nv-debuild-base"
+docker tag ${DOCKER_BASE} ${DOCKER_BASE_TAG}
+if [[ "$(docker run --rm ${DOCKER_BASE_TAG} bash -c "ldconfig -p | grep 'libcudart.so ' | wc -l")" -ne 1 ]]; then
+    echo "CUDA is not installed on this image"
+    exit 1
+fi
+if [[ "$(docker run --rm ${DOCKER_BASE_TAG} bash -c "ldconfig -p | grep 'libcudnn.so ' | wc -l")" -ne 1 ]]; then
+    echo "cuDNN is not installed on this image"
+    exit 1
+fi
+if [[ "$(docker run --rm ${DOCKER_BASE_TAG} bash -c "ldconfig -p | grep 'libnccl.so ' | wc -l")" -ne 1 ]]; then
+    echo "NCCL is not installed on this image"
+    exit 1
+fi
+
+if [ -z "$DEBIAN_REVISION" ]; then
+    echo ">>> Using default DEBIAN_REVISION (set the envvar to override)"
+    OS_NAME=$(docker run --rm ${DOCKER_BASE_TAG} sh -c ". /etc/os-release && echo \$ID\$VERSION_ID")
+    CUDA_VERSION=$(docker run --rm ${DOCKER_BASE_TAG} bash -c "readlink /usr/local/cuda-*/lib*/libcudart.so | sort -u | cut -c14-")
```

Comment:
I was avoiding it because I thought I might support images without CUDA. But realistically, all base images are going to come from nvidia-docker one way or another.